src/services/pcn-ddosmitigator/datamodel/ddosmitigator.yang

```diff
@@ -24,18 +24,30 @@ module ddosmitigator {
       type uint64;
       config false;
       description "Dropped Packets/s";
+      // polycube-base:name-metric "ddos_stats_pps_packets";
+      // polycube-base:type-metric "C";
```

Comment:
Can we define some enum (e.g., COUNTER, etc)?

Reply:
Hi @frisso , I don't know if I understand the comment well though, but playing a bit with the Yang of services (to add metrics) and seeing some examples, I don't think it is possible to define an enumeration within an extension but you should create an enumeration in each node of each yang such as:
```
        leaf state {
          type enumeration {
            enum disabled;
            enum blocking;
            enum listening;
            enum learning;
            enum forwarding;
          }
          description "STP port state";
          config false;
        }
```
I'm not Yang's top expert, maybe there are other solutions but at the moment I haven't found any (for example typedef). 

Always if I understand correctly, a possible solution that I have not tried yet but that does not require touching Yang could be to put an enum "inside" the struct InfoMetric and modify the parser with a function that takes a string and returns an enum, for example "COUNTER" taken from the yang. 

About this comment https://github.com/polycube-network/polycube/pull/313/#discussion_r462302489, doing so perhaps it would be possible to eliminate
```
if (metric.typeMetric == "C")
```
If it is not strictly necessary I would soon do it in another PR.

Reply:
Hi @frisso , I don't know if I understand the comment well though, but playing a bit with the Yang of services (to add metrics) and seeing some examples, I don't think it is possible to define an enumeration within an extension (the metrics are defined as extensions in polycube-base) but you should create an enumeration in each node of each yang such as:
```
        leaf state {
          type enumeration {
            enum disabled;
            enum blocking;
            enum listening;
            enum learning;
            enum forwarding;
          }
          description "STP port state";
          config false;
        }
```
I'm not Yang's top expert, maybe there are other solutions but at the moment I haven't found any (for example typedef). 

Always if I understand correctly, a possible solution that I have not tried yet but that does not require touching Yang could be to put an enum "inside" the struct InfoMetric and modify the parser with a function that takes a string and returns an enum, for example "COUNTER" taken from the yang. 

If it is not strictly necessary I would soon do it in another PR.

Reply:
Agreed, Giuseppe. Not strictly needed. Thanks!